### PR TITLE
Feature/job title

### PR DESCRIPTION
**Pull request by AmeerAmjed**
Handel :-
1-  Add job title.
2- Get all job titles.
3- Search in job title by id, title.
4- Delete job title by id

### CODE REVIEW

File: src/main/kotlin/com/the_chance/endpoints/JobTitle.kt [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    delete("/jobTitle/{idJobTitle}") {
+
+        suspend fun deleteJobTitle(idJobTitle: Int) {
+            jobTitleService.deleteJobTitleById(idJobTitle).takeIf { it }?.let { result ->
+                call.respond(HttpStatusCode.OK, ServerResponse.success(result))
+            } ?: call.respond(HttpStatusCode.NotFound, ServerResponse.error("Opps!, this job title not found."))
+        }
+
+        call.parameters["idJobTitle"]?.let { id ->
```
Comment (mustafa-sab3alarab) [inline src/main/kotlin/com/the_chance/endpoints/JobTitle.kt:52]: jobTitleId

File: src/main/kotlin/com/the_chance/endpoints/JobTitle.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    }
+
+    get("/jobTitles") {
+        jobTitleService.getAllJobTitle().takeIf { it.isNotEmpty() }?.let { posts ->
+            call.respond(HttpStatusCode.OK, ServerResponse.success(posts))
+        } ?: call.respond(HttpStatusCode.NoContent)
+    }
+
+    delete("/jobTitle/{idJobTitle}") {
```
Comment (mustafa-sab3alarab) [inline src/main/kotlin/com/the_chance/endpoints/JobTitle.kt:44]: jobTitleId

File: src/main/kotlin/com/the_chance/endpoints/JobTitle.kt [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            title.takeIf { validationTitle(it) == null }?.let {
+                handleJobTitleResult(createJobTitle(title))
+            } ?: call.respond(HttpStatusCode.BadRequest, ServerResponse.error(
+                    validationTitle(title) ?: "Failed to create job title")
+            )
+        } ?: call.respond(HttpStatusCode.BadRequest, ServerResponse.error("Title is required"))
+
+    }
+
+    get("/jobTitles") {
```
Comment (mustafa-sab3alarab) [inline src/main/kotlin/com/the_chance/endpoints/JobTitle.kt:38]: jobTitle

**Review by Karrar-Mohammed — APPROVED** [review]
well done 👏

**Review by mustafa-sab3alarab — APPROVED** [review]
well done

**Pull request merged**